### ISSUE TITLE
Method must be called

### ISSUE THREAD

**Issue by subzero911** [dart-lang/linter#2113]
**Describe the rule you'd like to see implemented**
Add an annotation for method that must be called at least once from somewhere and linter warning if it's not the case.

**Examples**
If I do not close a StreamSink, linter warns:
![image](https://user-images.githubusercontent.com/12999702/83195128-6903ed80-a142-11ea-9820-65c8b1e74c15.png)
If I close it in dispose() but do not call _bloc.dispose() from my StatefulWidget's dispose(), linter doesn't notice it.
![image](https://user-images.githubusercontent.com/12999702/83195171-791bcd00-a142-11ea-8a92-aab68176ea89.png)
Thus we easily can have memory leaks and never know about it. That's why I propose such a rule.

**Comment by PlugFox** [comment]
```dart
void main() {
  final A a = A();
  // ...
}

class A extends Sink<int> {
  @override
  void add(int data) {
    
  }

  @override
  void close() {
    
  }
}
```

**Comment by subzero911** [comment]
Not exactly what I wanted to.
I don't need add() method, because I do not add anything.
I need something like Closeable / Disposable.

**Comment by pq** [comment]
Without global flow analysis a lint like this would be hard to enforce.  (Note the various short-comings in `close_sinks`...)  That said, I love the idea of generalizing so we should discuss.  

See also: #2114 

/fyi @bwilkerson

**Comment by bwilkerson** [comment]
Unfortunately, that would also require a global analysis. The only performant checks we can make while you're typing are those that are local to the code being edited. That means that we can check, for example, that an object created in a method and not passed out of that method is also disposed in the same method. But as soon as a reference to the object escapes the method then we can't check to see whether it's disposed elsewhere so we can't know whether it's a problem if it isn't disposed in the same method.

**Comment by subzero911** [comment]
So, for my case I see two ways:
- To extend/implement Sink and leave add() empty or to throw an Exception('This class is not meant to add anything') in add method.
- Not to implement Sink and try not to forget that I need to close it (in Provider dispose slot or in Stateful dispose method)

**Comment by subzero911** [comment]
My goal is to have a `close_sinks` warning but not to implement add() method or to prevent calling it (just came up with workaround: mark add() method with @deprecated, so it will appear in Intellisense as striked out)
